patches/tModLoader/Terraria/ModLoader/IO/TileIO.cs

```diff
@@ -265,53 +458,93 @@ internal struct TileTables
 					tile.frameX = -1;
 					tile.frameY = -1;
 				}
-				if (tile.type == ModContent.Find<ModTile>("ModLoader/PendingUnloadedTile").Type
-					&& tables.tileNames.ContainsKey(saveType)) {
-					UnloadedTileInfo info;
-					if (tables.frameImportant[saveType]) {
-						info = new UnloadedTileInfo(tables.tileModNames[saveType], tables.tileNames[saveType],
-							tile.frameX, tile.frameY);
+
+				if ((flags & TileIOFlags.TileColor) == TileIOFlags.TileColor) {
+					tile.color(reader.ReadByte());
+				}
+
+				byte chestStyle = 0;
+				if ((flags2 & TileIOFlags2.ChestStyle) == TileIOFlags2.ChestStyle){
+					chestStyle = reader.ReadByte();
+				}
+
+				
+				if (tables.tileNames.ContainsKey(saveType)) {
+					// Handle Disabled Mods and Implement Pending Tile
+					bool nonSolidChk = tile.type == PendingNonSolidTile;
+					bool semiSolidChk = tile.type == PendingSemiSolidTile;
+
+					if (tile.type == PendingTile || nonSolidChk || semiSolidChk) {
+						// Load saved Basic Tile Type Data into UnloadedTileInfo and index
+						tInfo = new UnloadedTileInfo(tables.tileModNames[saveType], tables.tileNames[saveType],tables.tileFallback[saveType]);
+						posIndexer.SaveTileInfoToPos(tInfo);
+						tile.type = UnloadedTilesSystem.UnloadedTile;
+						if (nonSolidChk)
+							tile.type = UnloadedTilesSystem.UnloadedNonSolidTile;
+						if (semiSolidChk)
+							tile.type = UnloadedTilesSystem.UnloadedSemiSolidTile;
 					}
-					else {
-						info = new UnloadedTileInfo(tables.tileModNames[saveType], tables.tileNames[saveType]);
+					if (tile.type == PendingChest) {
+						bool accountedFor = ( //Extremely janky check that not sure why is needed and works
+							Main.tile[i - 1, j].type == UnloadedTilesSystem.UnloadedChest ||
+							Main.tile[i, j - 1].type == UnloadedTilesSystem.UnloadedChest ||
+							Main.tile[i - 1, j - 1].type == UnloadedTilesSystem.UnloadedChest);
+						if (!accountedFor) { // Only care about top-left tile of 2x2 for Chests
+							// Load saved Basic Chest Type Data into UnloadedChestInfo and index
+							UnloadedChestInfo cInfo = new UnloadedChestInfo(tables.tileModNames[saveType], tables.tileNames[saveType], chestStyle);
+							posIndexer.SaveChestInfoToPos(cInfo);
+							// Place UnloadedChest (required to preserve the inventory and re-namings
+							WorldGen.PlaceChestDirect(i, j + 1, UnloadedTilesSystem.UnloadedChest, 0, -1);
+						}
 					}
-					UnloadedTilesSystem unloadedTiles = ModContent.GetInstance<UnloadedTilesSystem>();
-					int pendingFrameID = unloadedTiles.pendingInfos.IndexOf(info);
-					if (pendingFrameID < 0) {
-						pendingFrameID = unloadedTiles.pendingInfos.Count;
-						unloadedTiles.pendingInfos.Add(info);
+					if (tile.type == PendingDresser) {
```

Comment:
`else if` perhaps?

Reply:
Agreed. All order-sensitive stuff is dealt with, so good catch.